dashboard/lib/services/curriculum_pdfs/resources.rb

```diff
@@ -191,14 +207,42 @@ def fetch_url_to_path(url, path)
           return nil
         end
 
-        # Returns a GoogleDrive::File object for the given url
-        #
-        # @see https://www.rubydoc.info/gems/google_drive/GoogleDrive/File
-        def google_drive_file_by_url(url)
-          @google_drive_session ||= GoogleDrive::Session.from_service_account_key(
-            StringIO.new(CDO.gdrive_export_secret&.to_json || "")
-          )
-          return @google_drive_session.file_by_url(url)
+        # # Returns a file_id for a google drive doc
+        # # borrowed from GoogleDrive api
+        # # See: https://github.com/gimite/google-drive-ruby/blob/55b996b2c287cb0932824bf2474248a498469328/lib/google_drive/session.rb#L693C5-L731C8
+        def url_to_id(url)
+          uri = URI.parse(url)
+          if ['spreadsheets.google.com', 'docs.google.com', 'drive.google.com'].include?(uri.host)
```

Comment:
I'm not sure we'd ever want to generate a PDF of a spreadsheet so we might be able to clean that code up for simplicity? Up to you though.

Reply:
I thought that too, but then I wanted to try to keep as much of the same old functionality as possible.  Since this is how the old code worked, I decided to keep it as is.  I think it would be rare for a curriculum writer to upload a spreadsheet as a PDF to include in the resources, but perhaps safer to keep it in than to take it out just in case any old resources are spreadsheets?

Reply:
Sound good to me!

Reply:
+1 on this, thinking long term, we would not want to slow down curriculum writers in the future if they decide to add spreadsheets to any lesson when the functionality was already in the code.